docs/docs-beta/docs/guides/build/project-structure/creating-a-component.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "Creating a New Component Type"
+sidebar_position: 100
+---
+
+:::note
+Refer to the project structuring guide to learn how to create a components-compatible project.
+:::
+
+The `dagster-components` system makes it easy to create new component types that can be reused across your project.
+
+In most cases, component types map to a specific technology. For example, you might have a `DockerScriptComponent` that executes a script in a Docker container, or a `SnowflakeQueryComponent` that runs a query on Snowflake.
+
```

Comment:
Do we not need a description (or a link) of how you make a python module a component library?